src/key.js

```diff
@@ -996,6 +1082,50 @@ function SubKey(subKeyPacket) {
   this.revocationSignatures = [];
 }
 
+/**
+ * generate a new SubKey.
+ * @param {Integer} options.numBits (optional) number of bits for the key creation.
+ * @param  {String} options.curve (optional) elliptic curve for ECC keys
+ * @param  {Date} options.date  Override the creation date of the key and the key signatures
+ *
+ * @returns {Promise<module:key.SubKey>} return the subkey if successful.
+ * @async
+ */
+SubKey.generate = async function(options) {
+  const result = new SubKey();
+  result.subKey = await generateSecretSubkey(options);
+  return result;
+};
+
+/**
+ * create the subkey binding signature
+ * @param  {module:packet.SecretKey} primaryKey The primary key packet
+ * @param  {Boolean} options.sign (optional) defaults to false, and indicates whether the subkey should sign rather than encrypt
+ * @param {Number} [options.keyExpirationTime=0]
+ *                             The number of seconds after the key creation time that the key expires
+ * @param  {Date} options.date  Override the creation date of the key and the key signatures
+ * @param  {Object} userId                   (optional) user ID
+ *
+ * @returns {Promise<module:packet.Signature>} return the signature if successful.
+ * @async
+ */
+SubKey.prototype.bindSignature = async function(primaryKey, options) {
+  const subkeySignatureOpts = {};
```

Comment:
Please find a way to share this code with wrapKeyObject, as most of it is duplicated from there